google_guest_agent_packaging_diff_tool.py

```diff
@@ -1,46 +1,111 @@
-from pprint import pprint
 import os
 import click
+import pprint
+import requests
+import warnings
+
+from dataclasses import dataclass
 from debian import deb822
+from debian.deb822 import Deb822
+
+UBUNTU_CONTROL_FILENAME = "ubuntu_control.file"
+GOOGLE_CONTROL_FILENAME = "gce_control.file"
+
+
+@dataclass
+class ArchiveURLS:
+    ubuntu_control_url: str
+    google_upstream_control_url: str
+
+archives: dict[str, ArchiveURLS]
+
+archive_dict = {
+    "google-guest-agent": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-guest-agent/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-agent/main/packaging/debian/control"
+    ],
+    "gce-compute-image-packages": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/gce-compute-image-packages/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-configs/master/packaging/debian/control"
+    ],
+    "google-compute-engine-oslogin": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-compute-engine-oslogin/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-oslogin/master/packaging/debian/control"
+    ],
+    "google-osconfig-agent": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-osconfig-agent/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/osconfig/master/packaging/debian/control"
+    ],
+} # type: archives
+
+def clean_up() -> None:
+    os.remove(UBUNTU_CONTROL_FILENAME)
+    os.remove(GOOGLE_CONTROL_FILENAME)
+
+def parse_control_file(control_file: str | os.PathLike) -> Deb822:
+    with open(control_file) as fp:
+        control_file_text = fp.read()
 
-def parse_control_file(control_file):
-    control_file_text = control_file.read()
     control_file_text_no_blank_lines = os.linesep.join([s for s in control_file_text.splitlines() if s])
     control_fileds = deb822.Deb822(control_file_text_no_blank_lines)
```

Comment:
You could shorten `deb822.Deb822` to `Deb822` since you've imported that name.

Reply:
Done, it's in the next upload :smile: